File: 백준/Bronze/4583. 거울상/거울상.py
while(True):
  a = input()
  if a == '#':
    break

  result = []
  for i in range(len(a) - 1, -1, -1):
    if 'i' == a[i] or 'v' == a[i] or 'o' == a[i] or 'w' == a[i] or 'x' == a[i]:
      result.append(a[i])
    elif a[i] == 'b':
      result.append('d')
    elif a[i] == 'd':
      result.append('b')
    elif a[i] == 'p':
      result.append('q')
    elif a[i] == 'q':
      result.append('p')
    else:
      break

  if len(result) != len(a):
    print("INVALID")
  else:
    for item in result:
      print(item, end = '')
    print()

File: 백준/Bronze/4583. 거울상/거울상.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   char letter[11];
   while(1){
      char result[11] = {0};
      int i = 0, index = 0;
      scanf("%s", letter);

      if(strcmp(letter, "#") == 0)   break;

      for(i = strlen(letter) - 1; i >= 0; i--){
         if(letter[i] == 'i' || letter[i] == 'o' || letter[i] == 'w' || letter[i] == 'v' || letter[i] == 'x')
            result[index] = letter[i];

         else if(letter[i] == 'b')   result[index] = 'd';
         else if(letter[i] == 'd')   result[index] = 'b';
         else if(letter[i] == 'p')   result[index] = 'q';
         else if(letter[i] == 'q')   result[index] = 'p';
         else   break;

         index++;
      }

      result[strlen(letter)] = '\0';
      if(i >= 0){
         printf("INVALID\n");
      }
      else{
         printf("%s\n", result);
      }
      
   }
   
   return 0;
}